macros/upload_results.sql

```diff
@@ -126,14 +126,22 @@
 
         {% do log("Uploading sources", true) %}
         {% set sources = dbt_artifacts.get_relation('sources') %}
-        {% set content_sources = dbt_artifacts.upload_sources(graph) %}
-        {{ dbt_artifacts.insert_into_metadata_table(
-            database_name=sources.database,
-            schema_name=sources.schema,
-            table_name=sources.identifier,
-            content=content_sources
-            )
-        }}
+        {% set sources_set = [] %}
+        {% for node in graph.nodes.values() | selectattr("resource_type", "equalto", "source") %}
+            {% do sources_set.append(node) %}
+        {% endfor %}
+        {# upload sources in chunks of 500 sources (300 for BigQuery), or less #}
+        {% set upload_limit = 300 if target.type == 'bigquery' else 500 %}
+        {% for i in range(0, sources_set | length, upload_limit) -%}
+            {% set content_sources = dbt_artifacts.upload_sources(sources_set[i: i + upload_limit]) %}
+            {{ dbt_artifacts.insert_into_metadata_table(
+                database_name=sources.database,
+                schema_name=sources.schema,
+                table_name=sources.identifier,
+                content=content_sources
+                )
+            }}
+        {%- endfor %}
```

Comment:
When making this change you will also need to make a change `upload_sources` macro.

And [example of this can be found in this merged PR](https://github.com/brooklyn-data/dbt_artifacts/commit/b7d45ac88ed6f882c97cbf9d408b6a9167895dd7)

Reply:
Done in the next commit